lib/launcher.rb

```diff
@@ -12,18 +25,25 @@ class Launcher
 
     def initialize(settings = SETTINGS)
       @settings = settings
+      @settings.http_server_type = Proxy::SETTINGS.http_server_type.to_sym
+      @settings.https_server_type = Proxy::SETTINGS.https_server_type.to_sym
+      if @settings.http_server_type == :puma && !$HAS_PUMA
```

Comment:
I don't think we need an extra `$GLOBAL_VARIABLE`, we can just check on `defined? Puma`

Reply:
my issue with `define puma` is that if something else regarding puma was not loaded we do not know of it, and then looking for every point that is needed is a bit of an issue in my opinion.

You will end up either writing a function that does this, or have duplicate places that check that code.

So knowing that everything was loaded properly, even on other locations at the project when needed is a bit easier in my opinion and easier for the long run.